src/pages/LoginWithValidateCode2FAPage.js

```diff
@@ -58,21 +61,44 @@ class ValidateLogin2FANewWorkspacePage extends Component {
     }
 
     componentDidMount() {
-        // If the user has an active session already, they need to be redirected straight to the new workspace page
+        // If the user has an active session already, they need to be redirected straight to the relevant page
         if (this.props.session.authToken) {
             // In order to navigate to a modal, we first have to dismiss the current modal. But there is no current
-            // modal you say? I know, it confuses me too. Without dismissing the current modal, if they user cancels
-            // out of the new workspace modal, then they will be routed back to
-            // /v/<accountID>/<validateCode>/new-workspace and we don't want that. We want them to go back to `/` and
-            // by calling dismissModal(), the /v/... route is removed from history so the user will get taken to `/`
+            // modal you say? I know, it confuses me too. Without dismissing the current modal, if the user cancels out
+            // of the new workspace modal, then they will be routed back to
+            // /v/<accountID>/<validateCode>/workspace/123/card and we don't want that. We want them to go back to `/`
+            // and by calling dismissModal(), the /v/... route is removed from history so the user will get taken to `/`
             // if they cancel out of the new workspace modal.
             Navigation.dismissModal();
+            if (Permissions.canUseFreePlan(this.props.betas)) {
+                this.rerouteToRelevantPage();
+            }
+        }
+    }
 
-            if (_.isEmpty(this.props.betas)) {
-                setRedirectToWorkspaceNewAfterSignIn(true);
-            } else {
+    componentDidUpdate() {
+        // Betas can be loaded a little after a user is authenticated, so check again if the betas have been updated
+        if (this.props.session.authToken && Permissions.canUseFreePlan(this.props.betas)) {
+            this.rerouteToRelevantPage();
+        }
+    }
+
+    rerouteToRelevantPage() {
+        // Since all 2FA validate code login routes lead to this component, redirect to the appropriate page based on
+        // the original route.
+        switch (this.props.route.name) {
+            case SCREENS.LOGIN_WITH_VALIDATE_CODE_2FA_WORKSPACE_CARD:
+                Navigation.navigate(ROUTES.getWorkspaceCardRoute(this.props.route.params.policyID));
+                break;
+
+            case SCREENS.LOGIN_WITH_VALIDATE_CODE_2FA_NEW_WORKSPACE:
+                // Creating a policy will reroute the user to the settings page afterwards
                 create();
```

Comment:
Seeing this in the surrounding context, it's hard to know what this does. I would suggest changing the import to be `import * as Policy from....` then change this reference to `Policy.create()` so that it helps add context about what's getting created without having to check the imports.